core/src/main/java/io/aiven/kafka/tieredstorage/config/CacheConfig.java

```diff
@@ -38,59 +38,21 @@ public class CacheConfig extends AbstractConfig {
     private static final String CACHE_FETCH_TIMEOUT_MS_DOC = "When getting an object from the fetch, "
         + "how long to wait before timing out. Defaults to 10 sec.";
 
-    static final long CACHE_RETENTION_MS_DEFAULT = 600_000;
+    public static final long CACHE_RETENTION_MS_DEFAULT = 600_000;
 
-    private static ConfigDef configDef(
+    public CacheConfig(
         final ConfigDef configDef,
-        final Object defaultSize,
-        final long defaultRetentionMs
+        final Map<String, ?> props
     ) {
```

Comment:
Disentangle configDef from CacheConfig. Instead the `DefBuilder` defines the config definition.